pkg/kube/client.go

```diff
@@ -149,8 +150,8 @@ func (c *Client) Build(reader io.Reader, validate bool) (ResourceList, error) {
 // occurs, a Result will still be returned with the error, containing all
 // resource updates, creations, and deletions that were attempted. These can be
 // used for cleanup or other logging purposes.
-func (c *Client) Update(original, target ResourceList, force bool) (*Result, error) {
-	updateErrors := []string{}
+func (c *Client) Update(original, target ResourceList, force bool, recreate bool, timeout time.Duration) (*Result, error) {
```

Comment:
This would be considered an API change to the SDK. In general we try to avoid these. Is there a way we can refactor this such that this function signature remains the same?

Reply:
Changes should be non breaking now, although the solution is not very nice. PR is ready for final review.